packages/kit/types/index.d.ts

```diff
@@ -71,6 +71,22 @@ type UnpackValidationError<T> = T extends ActionFailure<infer X>
 	? undefined // needs to be undefined, because void will corrupt union type
 	: T;
 
+export type ActionsSuccess<T extends Record<string, (...args: any) => any>> = {
+	[Key in keyof T]: ExcludeValidationError<Awaited<ReturnType<T[Key]>>>;
+}[keyof T];
+
+type ExcludeValidationError<T> = T extends ActionFailure<any> ? never : T extends void ? never : T;
+
+export type ActionsInvalid<T extends Record<string, (...args: any) => any>> = {
+	[Key in keyof T]: Exclude<ExtractValidationError<Awaited<ReturnType<T[Key]>>>, void>;
+}[keyof T];
+
+type ExtractValidationError<T> = T extends ActionFailure<infer X>
+	? X extends void
+		? never
+		: X
+	: never;
```

Comment:
I'm debating whether or not it would be better move this into the helper types in `$types` to not bloat the public types, or if we keep it here, if there's room to condense these into one more generic helper.

Reply:
The question is: should we allow anyone to manually type the `SubmitFunction`?
If yes, those types need to stay here, else we can move them to the generated `$types.d.ts`.

I was also thinking about refactoring the types, but it will make them probably harder to read if we add an abstraction..

Reply:
What should I do here? Leave it or move it to `$types.d.ts`?

Reply:
Let's move it into `$types.d.ts` for now

Reply:
done